File: docs/graphify-batches/corpora/workers-real/ave/src/web/studio/src/types/schemas.ts
/** Mirrors src/models/schemas.py exactly. */

export interface CreativeBrief {
  product: string;
  audience: string;
  tone: string;
  duration_seconds: number;
  style_ref: string | null;
}

export interface WordTimestamp {
  word: string;
  start: number;
  end: number;
}

export interface Shot {
  source_file: string;
  start_time: number;
  end_time: number;
  description: string;
  energy_level: number;
  relevance_score: number;
  transcript: string;
  words: WordTimestamp[];
  roll_type: string;
}

export interface FootageIndex {
  source_dir: string;
  shots: Shot[];
  total_duration: number;
  created_at: string;
}

export interface EditPlanEntry {
  shot_id: string;
  start_trim: number;
  end_trim: number;
  position: number;
  text_overlay: string | null;
  transition: string | null;
}

export interface EditPlan {
  brief: CreativeBrief;
  entries: EditPlanEntry[];
  music_path: string | null;
  total_duration: number;
}

export interface ReviewScore {
  adherence: number;
  pacing: number;
  visual_quality: number;
  watchability: number;
  overall: number;
  feedback: string;
}

File: docs/graphify-batches/corpora/workers-real/ave/src/models/schemas.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class CreativeBrief(BaseModel):
    product: str
    audience: str
    tone: str
    duration_seconds: int
    style_ref: str | None = None


class WordTimestamp(BaseModel):
    word: str
    start: float
    end: float


class Shot(BaseModel):
    source_file: str
    start_time: float
    end_time: float
    description: str
    energy_level: int
    relevance_score: float
    transcript: str
    words: list[WordTimestamp] = Field(default_factory=list)
    roll_type: str = Field(
        default="unknown",
        description=(
            "Footage category: 'a-roll' (on-camera talent / talking head), "
            "'b-roll' (cutaway / product / texture / environment), or "
            "'unknown'. Auto-detected from the source directory name during "
            "preprocessing."
        ),
    )


class FootageIndex(BaseModel):
    source_dir: str
    shots: list[Shot]
    total_duration: float
    created_at: datetime


class EditPlanEntry(BaseModel):
    shot_id: str
    start_trim: float
    end_trim: float
    position: int
    text_overlay: str | None = None
    transition: str | None = None


class EditPlan(BaseModel):
    brief: CreativeBrief
    entries: list[EditPlanEntry]
    music_path: str | None = None
    total_duration: float


class ReviewScore(BaseModel):
    adherence: float
    pacing: float
    visual_quality: float
    watchability: float
    overall: float
    feedback: str
